lib/schmooze/base.rb

```diff
@@ -44,15 +44,18 @@ def initialize(root, env={})
       @env = env
       @root = root
       @code = ProcessorGenerator.generate(self.class.instance_variable_get(:@imports) || [], self.class.instance_variable_get(:@methods) || [])
-
-      spawn_process
     end
 
     def pid
-      @process_thread.pid
+      @process_thread && @process_thread.pid
     end
 
     private
+      def ensure_process_spawned
+        return if @process_thread
```

Comment:
I do care if it dies while getting the dependencies, fixed that and added a test for it, thanks